### ISSUE TITLE
[Invoices] Epic: Module Invoices

### ISSUE THREAD

**Issue by naui95** [InvoicePlane/InvoicePlane-v2#15]

**Comment by nielsdrost7** [comment]
## SMART Plan — Invoices module epic status

> **Context/Epic**: Invoices
>
> This is an epic/tracking issue, not a single feature — it aggregates the sub-issues below. The Measurable section tracks sub-issue completion rather than fabricating tests for the epic itself; each sub-issue gets its own SMART Plan when it's processed individually.

### Specific
Track delivery of the Invoices module: the core `Invoice`/`InvoiceItem` CRUD and recurring-invoice scaffolding are done, but most of the lifecycle actions (email, copy, credit notes, PDF, inline payment), grouping/filtering, and the recurring-invoice automation itself are still unimplemented stubs.

### Measurable

**Actions & lifecycle**
- [ ] #150 — Email invoice to customer — **not started**. `InvoicesTable`'s "send email" action calls `InvoiceService::sendInvoiceEmail()`, which doesn't exist anywhere in the codebase; clicking it today throws a fatal error. Broken down into #522/#524/#526/#527/#529/#531 in this batch.
- [ ] #167 — Copy/duplicate an invoice — **not started**. `Modules\Invoices\Services\InvoiceCopyService` is an empty stub class.
- [ ] #171 — Issue a credit note from a paid invoice — **not started**. `Modules\Invoices\Services\CreditInvoiceService` is an empty stub, though the schema groundwork (`creditinvoice_parent_id`, `invoice_sign`) already exists.
- [ ] #190 — Download PDF from the invoice edit/detail view — **not started**. The "download pdf" `Action` in `InvoicesTable` exists but its `->action()` callback is empty (`function (Invoice $record): void {}`) and its own `modalDescription` literally says "todo".
- [ ] #200 — Download PDF from the invoice list row — **not started**, same empty stub action as #190.
- [ ] #201 — Enter payment inline from the invoice list — **not started**, no such action found anywhere.
- [ ] #223 — Action dropdown on invoice edit page (Draft state) — **partially started**. The `ActionGroup` (Edit / download pdf / send email / delete) already exists on the list, but nothing status-conditional is wired in yet.
- [ ] #224 — Action dropdown on invoice edit page (Paid state — Delete hidden) — **not started**. `DeleteAction` today has no status-based visibility condition.
- [ ] #31 — Complete the invoice list action group — **partially started** (see #223); umbrella issue for #150/#190/#200/#201/#223/#224 together.

**Grouping & filtering**
- [ ] #46 — Move invoice from one numbering group to another — **not started**. `InvoicesTable::configure()` has `->filters([])`, empty; no move action exists.
- [ ] #47 — Reports per invoice group — **not started**.
- [ ] #48 — Filter invoice list by invoice group — **not started**. (#45 in this batch is community feedback that overlaps with these three.)

**Credit notes**
- [ ] #42 — Read-only state when a credit note exists — **not started**. `is_read_only` column exists on `invoices` but nothing sets it based on credit-note existence.
- [ ] #44 — Deletion guard when a credit note exists — **not started**. `InvoiceService::deleteInvoice()` currently only guards against deleting a `PAID` invoice, nothing credit-note-specific.
- [ ] #367 — Settlement of balances for credit invoices — **not started** (`CreditInvoiceService` stub; case 3 in this batch).

**Recurring invoices**
- [ ] #52 — Full recurring invoices lifecycle — **not started**. `Modules\Invoices\Services\RecurringInvoiceService` is an empty stub; `RecurringInvoicesTable` only has Edit/Delete; `RecurringInvoicesTest` is entirely `markTestIncomplete()` with undefined-variable bugs; the model/factory still reference a `DocumentGroup` class that no longer exists in the codebase (renamed to `Numbering`).
- [ ] #127 — Recurring invoices dashboard widget — **not started**.
- [ ] #143 — Filter invoice list to show only recurring invoices — **not started**; also needs a new `recurring_invoice_id` column on `invoices` that doesn't exist today (case 1 in this batch).
- [ ] #362 — Re-activate a stopped recurring invoice — **not started**; there is no "stop" mechanism in v2 at all yet to reactivate from (case 3 in this batch).
- [ ] #368 — Auto-assign invoice number to draft recurring invoices — **not started**; blocked on #52 existing at all (case 3 in this batch).

**Other**
- [ ] #135 — Tax calculation correctness tests — **partially available**. `InvoiceCalculator` (`Modules/Invoices/Support/InvoiceCalculator.php`) has real, non-stub tax/discount/total calculation logic already, but no dedicated test file for it exists yet (`Modules/Invoices/Tests` has no `*Tax*` test).
- [ ] #480 — Invoice reminders for overdue invoices — **not started**; also blocked on a client-email resolution gap (`Relation` has no real `email` column — see #480's own grounding in this batch) and on the same `MailQueue`/PDF pipeline as #150/#411.
- [ ] #370 — Consecutive line-item numbering — **not started**. `invoice_items.display_order` is passed through from input as-is (`$item['display_order'] ?? null`) with no auto-increment logic in `InvoiceService`.
- [ ] #403 — Output invoice with a "copy" watermark — **not started**; depends on #130's PDF pipeline landing (case 3 in this batch).

**Definition of done for this epic**: all sub-issues above are closed, `Modules/Invoices/Tests/Feature/` covers each new capability (including fixing `RecurringInvoicesTest`'s existing `markTestIncomplete()` calls), and the module no longer contains dead references (`DocumentGroup`) or fatal stub calls (`InvoiceService::sendInvoiceEmail()`).

### Achievable
1. #150 (email) and #52 (recurring lifecycle) are the two foundational pieces most other sub-issues build on — #150 unblocks #480 and the "send" half of #403; #52 unblocks #127, #143, #362, #368.
2. #171 (issue a credit note) should land before #42/#44/#367, since those three all guard around or settle against a credit note's *existence* — right now there's no way to create one to guard around.
3. #46/#48 (grouping/filtering by numbering group) have no dependency on anything else in this epic and can be picked up immediately; #47 (reports per group) naturally follows #48's filtering work.
4. #190/#200 (download PDF) and #403 (copy watermark) all depend on `Modules\Core\Services\PdfGenerationService` being populated — it's currently an empty stub in `develop`, but a complete, tested implementation already exists on the unmerged `feature/130-report-builder` branch (see #411 in this batch), so this is a merge dependency, not new-build work.
5. #223/#224 (status-conditional action visibility) and #31 (the umbrella) should land last, once #150/#190/#200/#201 give the action group something real to condition on.

### Relevant
Invoices is the core billing module — its lifecycle actions (email, PDF, credit notes, recurring automation) are what customers actually interact with daily, and right now most of that surface area is either an empty stub or a call to a method that doesn't exist.

### Time-bound
No fixed date — epic closes when its sub-issues close. Re-review at each sprint boundary; #150 and #52 should be prioritized first since the most other work depends on them.

### Out of scope (unchanged)
- Multi-currency invoicing beyond the existing `currency`/`exchange_rate` fields
- Custom per-client invoice numbering formats beyond the existing `Numbering` model
